### PR TITLE
LPS-163157 Version pagination in info panel

### PR DESCRIPTION
**Pull request by jkappler**
## Motivation

[Link to story](https://issues.liferay.com/browse/LPS-163157)

Tango Team is working in improving the information panel in content dashboard, and one of these improvements is adding the version history of the elements in the sidebar. Since an element can have many versions, we would like to provide the user the ability to paginate through them.

## Proposed Solution

Create an endpoint that can be called with a page number and the response includes the total pages for the item, and the versions for the given page

## Change summary:

* Extract version methods to a new interface called VersionableContentDashboardItem
* Remove versions from the initial payload of info panel
* Create endpoint to retrieve versions

## TODO

- [ ] Add integration tests
- [ ] Adapt frontend test 
- [x] Remove allVersions method since it will no longer be used
- [x] Adapt frontend to use the endpoint

### CODE REVIEW

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.List;
+
+/**
+ * @author Cristina González
+ */
+public interface VersionableContentDashboardItem<T>
+	extends ContentDashboardItem<T> {
+
+	public List<ContentDashboardItem.Version> getAllVersions(
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java:27]: Does that method make sense anymore?

Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java:27]: Hi @cgoncas, no, it's the method that we'll remove

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java:27]: Totally true!!!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionsContent.tsx [hunk 2]
```diff
@@ -32,42 +32,49 @@ const VersionsContent = ({
...
 
-				if (process.env.NODE_ENV === 'development') {
-					console.error(error);
-				}
-			}
-			finally {
-				setLoading(false);
-			}
-		};
+	useEffect((): void => {
```
Comment (beltranrengifo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionsContent.tsx:66]: Hi @claraizquierdo !
After giving some thoughts to this this problem, I think we can simplify just using the `useIsFirstRender` hook and cancelling the fetch in that case, this way:

```
	useEffect(() => {
		if (isFirst) {
			return;
		}
		getVersionsData();
	}, [getVersionsData, isFirst]);
```

If I'm not mistaken, this way we avoid the initial unnecessary fetch, but get the data on first tab and page changes.
I'll push some changes.
Do you mind to give it a try?
Thxs!

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java [hunk 3]
```diff
@@ -24,9 +24,6 @@
 public interface VersionableContentDashboardItem<T>
 	extends ContentDashboardItem<T> {
 
-	public List<ContentDashboardItem.Version> getAllVersions(
```
Comment (claraizquierdo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java]: Hi @jkappler , that method is overwritten for File Entries and Journal Articles. Should we remove all of them?

Comment (jkappler) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/VersionableContentDashboardItem.java]: yes, thanks for reminding me

**Comment by jkappler** [comment]
We will continue here: https://github.com/liferay-tango/liferay-portal/pull/2842

**Pull request closed without merging**